src/Raven.Client/Http/NodeSelector.cs

```diff
@@ -66,12 +66,16 @@ public bool OnUpdateTopology(Topology topology, bool forceUpdate = false)
             var state = _state;
             var stateFailures = state.Failures;
             var serverNodes = state.Nodes;
-            var len = Math.Min(serverNodes.Count, stateFailures.Length);
-            for (var i = 0; i < len; i++)
+            
+            if (serverNodes.Count == 1 && serverNodes[0].ClusterTag == nodeTag) // If this is a cluster with 1 node return it without checking it's failure.
+                return (0, serverNodes[0]);
+
+            for (var i = 0; i < serverNodes.Count; i++)
             {
                 if (serverNodes[i].ClusterTag == nodeTag)
                 {
-                    if (stateFailures[i] == 0 && string.IsNullOrEmpty(serverNodes[i].Url) == false)
+                    if (i>= stateFailures.Length || // if the node doesn't have an element in stateFailure array, it's considered to have 0 failures
```

Comment:
This shouldn't happen since `state.Nodes.Count == state.Failures.Length`

Reply:
I removed it